lnbits/wallets/strike.py

```diff
@@ -0,0 +1,551 @@
+        except Exception:  # Handle other exceptions.
+            logger.exception("Error in get_invoice_status()")
+            return PaymentPendingStatus()
+
+    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
+        quote_id = self.pending_payments.get(checking_id)
+        if not quote_id:  # Payment not found in pending list.
+            if checking_id in self.failed_payments:
+                return PaymentFailedStatus(
+                    paid=False
+                )  # Payment is known to have failed.
+            return PaymentPendingStatus()
+        try:  # Try to get payment quote.
+            r = await self._get(
+                f"/payment-quotes/{quote_id}"
+            )  # Get payment quote from Strike API.
+            r.raise_for_status()
+            data = r.json()  # Parse JSON response.
+            state = data.get("state")  # Get payment state.
+            preimage = data.get("preimage") or data.get(
+                "preImage"
+            )  # Get payment preimage.
+            if state in ("SUCCEEDED", "COMPLETED"):  # If payment succeeded.
+                return PaymentSuccessStatus(
+                    fee_msat=0, preimage=preimage
+                )  # Return successful payment status.
+            if state == "PENDING":  # If payment is pending.
+                return PaymentPendingStatus()  # Return pending payment status.
+            return PaymentStatus(False)  # Return failed payment status.
+        except httpx.HTTPStatusError as e:
+            if e.response.status_code == 404:
+                # Quote not found, likely expired or payment failed.
+                self.pending_payments.pop(checking_id, None)
+                self.failed_payments[checking_id] = quote_id
+                return PaymentFailedStatus(paid=False)
+            raise  # Re-raise other HTTP errors
+        except Exception:  # Handle exceptions.
+            logger.exception(f"Error in get_payment_status() for payment {checking_id}")
+            return PaymentPendingStatus()  # Treat as pending for now.
+
+    async def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
+        """
+        Poll Strike for invoice settlement while respecting the documented API limits.
+
+        Uses dynamic adjustment of polling frequency based on activity.
+        """
+        min_poll, max_poll = 1, 15
+        # 1,000 requests / 10 minutes = ~100 requests/minute.
+        rate_limit = 100
+        sleep_s = min_poll
+        # Main loop for polling invoices.
+        self._running = True
+
+        while self._running and settings.lnbits_running:
+            loop_start = time.time()
+            had_activity = False
+
+            req_budget = max(
+                1, rate_limit * sleep_s // 60
+            )  # Calculate request budget based on sleep time.
+            processed = 0
+
+            for inv in list(self.pending_invoices):  # Iterate through pending invoices.
```

Comment:
After a server restart this list will be empty.

Reply:
LNbits does a regular check (every `30 min`) on `payments` like so:

```py
  funding_source = get_funding_source()
  if self.is_out:
      status = await funding_source.get_payment_status(self.checking_id)
  else:
      status = await funding_source.get_invoice_status(self.checking_id)
```

So these two functions need to make the right API calls, return the correct status and LNbits takes care of the rest.